doc/holochain_101/src/writing_development_kit.md

```diff
@@ -1 +1,100 @@
 # Writing a Development Kit
+
+The end goal of a Development Kit is to simplify the experience of writing Zomes that compile to WASM for Holochain apps.
+
+At the time of writing, there is currently one active Developer Kit being written, for the Rust language. While it is possible to look at the [Rust language HDK](https://github.com/holochain/hdk-rust) as a reference, this article is a more general guide that outlines what it takes to build a Development Kit.
+
+If you are interested in supporting developers to write Zomes in an unsupported language, you will want to first of all check whether that language can be compiled to WebAssembly, as that is a requirement.
+
+### Why Development Kits
+
+Development Kits are important because the WASM interface between Zomes and Holochain is really constrained. Because of WASMs design, WASM functions may only be called with 32 bit integers. Holochain implements a solution to this, but if app developers were to always have to interact with this solution directly, it would feel very complex. A Development Kit for each language should ideally be developed so that it gets so much simpler!
+
+### The Development Kit WASM Solution
+
+To enable passing arguments more complex than 32 bit integers between Zomes and Holochain, a pattern of utilizing WASM memory is used. When it is running the WASM code for a Zome, Holochain has access to both read and write from the WASM memory.
+
+The pattern defines that Holochain Zome API functions expect to both give and receive 32 bit integers which actually represent a WASM memory location. So to provide a Holochain Zome API function a complex argument, one must first write it into memory, and then call the function, giving it the memory location. Holochain will pull the argument from memory, execute its behaviour, store the result in memory, and return the memory location of the result. The Zome code then has to *also* lookup the result by its location in memory.
+
+Technically, an app developer can do all of these things if they have a reason to, but most won't want to handle the extra step involving memory. A Development Kit, then, should handle the extra step of writing to memory, and calling the native API function, and reading the result from memory, and returning that instead. Plus a few other sprinkles on top.
+
+### Crafting the API
+
+Using its WASM interpreter, Holochain exposes its callable Zome API functions by making them available as "imports" in Zome WASM modules. Per the memory discussion above, each of the Zome API functions have the same explicit function signature, but different implicit function signatures. The native functions have each been given a prefix so that Development Kit wrappers can expose a regular function name. Here is a complete list:
+
+- hc_debug
```

Comment:
I think you shouldn't actually put a complete list of the function here, instead we should point to them elsewhere, probably rust-spec repo, so that we don't have to remember to update it here all the time and things don't get out of sync.   But for now I'll just merge this PR.

Reply:
Agreed @zippy ... while we are still sorting out how we are going to publish the literal docs for the code itself, which in some cases will act as a great place to refer to